src/SFML/System/Win32/LibraryLoader.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_LIBRARY_LOADERWIN32_HPP
+#define SFML_LIBRARY_LOADERWIN32_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <SFML/System/NonCopyable.hpp>
+#include <SFML/System/String.hpp>
+#include <windows.h>
+
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+/// \brief todo
+///
+////////////////////////////////////////////////////////////
+class SFML_SYSTEM_API LibraryLoader : public NonCopyable
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    LibraryLoader() = delete;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// todo
+    ///
+    /// \param libraryName todo
+    ///
+    ////////////////////////////////////////////////////////////
+    LibraryLoader(const sf::String& libraryName);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default destructor
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    virtual ~LibraryLoader();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// \return todo
+    ///
+    ////////////////////////////////////////////////////////////
+    bool getIsLoaded();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// todo
+    ///
+    /// \param procName todo
+    ///
+    /// \return True if the TODO
+    ///
+    ////////////////////////////////////////////////////////////
+    template<typename TFuncType>
+    bool getProcAddress(TFuncType& procAddress, const sf::String& procName);
```

Comment:
Write out the full name of terms, i.e. `getProcedureAddress`

Reply:
```suggestion
    [[nodiscard]] bool getProcAddress(TFuncType& procAddress, const sf::String& procName);
```

Reply:
Done. Added [[nodiscard]]. Fixed shortened names for functions and some variables.